compiler/il/OMRIL.cpp

```diff
@@ -763,6 +560,11 @@ OMR::IL::opCodeForDirectLoad(TR::DataType dt)
 TR::ILOpCodes
 OMR::IL::opCodeForDirectReadBarrier(TR::DataType dt)
    {
+   static_assert(TR::NumOMRTypes == (sizeof(OMR::IL::opCodesForDirectReadBarrier) / sizeof(OMR::IL::opCodesForDirectReadBarrier[0])),
+              "OMR::IL::opCodesForDirectReadBarrier is not the correct size");
+
+   if (dt.isVector()) return TR::BadILOp;
```

Comment:
If a read barrier is really a "load with some extra semantics" should we allow a `vreadbarrier` (given that we have a `vload`) ?

Reply:
I see, thanks. So this may be worth changing in the future but I agree with preserving behavior in this PR.